### PR TITLE
Use PodTemplate metadata in managed Pods

### PR DESCRIPTION
**Pull request by hamersaw**
# TL;DR
Instead of initializing managed pods with a new ObjectMeta this PR resuses the ObjectMeta on the PodTemplate. This means that all managed FlytePropeller pods will adhere as closely to the original PodTemplate as possible, meaning labels, annotations, etc are set correctly.

This approach is proposed as an alternative to https://github.com/flyteorg/flytepropeller/pull/410.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
 ^^^

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by hamersaw** [comment]
@mayitbeegh can you pull this and make sure it works as you expect?

**Review by EngHabu — APPROVED** [review]

**Review by EngHabu — APPROVED** [review]
